public/app/features/dashboard/components/DashNav/DashNavTimeControls.tsx

```diff
@@ -14,23 +13,23 @@ import { TimePickerWithHistory } from 'app/core/components/TimePicker/TimePicker
 // Utils & Services
 import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
 import { appEvents } from 'app/core/core';
-import { ShiftTimeEvent, ShiftTimeEventPayload, ZoomOutEvent } from '../../../../types/events';
+import { ShiftTimeEvent, ShiftTimeEventPayload, TimeRangeUpdatedEvent, ZoomOutEvent } from '../../../../types/events';
+import { Unsubscribable } from 'rxjs';
 
 export interface Props {
   dashboard: DashboardModel;
   onChangeTimeZone: (timeZone: TimeZone) => void;
 }
 
 export class DashNavTimeControls extends Component<Props> {
+  sub?: Unsubscribable;
+
   componentDidMount() {
-    // Only reason for this is that sometimes time updates can happen via redux location changes
-    // and this happens before timeSrv has had chance to update state (as it listens to angular route-updated)
-    // This can be removed after timeSrv listens redux location
-    this.props.dashboard.on(CoreEvents.timeRangeUpdated, this.triggerForceUpdate);
+    this.sub = this.props.dashboard.events.subscribe(TimeRangeUpdatedEvent, () => this.forceUpdate());
```

Comment:
Is the previous comment not valid anymore? I think it was great.

Reply:
it's not really valid anymore as it pertained to redux location and angular route events.

Reply:
and we have none of those after v8